syntax = "proto3";

package comunicacao;

option go_package = "./pb";

service BIQueryService {
  rpc GetMarcaStats (Filtro) returns (MarcaStats);
  rpc GetContagemSegmento (Filtro) returns (Resultado);
  rpc GetLocalizacaoStats (Filtro) returns (LocalizacaoStats);
}

message Filtro {
  string termo = 1;
}

message Resultado {
  float valor = 1;
}

message MarcaStats {
  int32 total = 1;
  float media_preco = 2; // O Go vai ler isto como MediaPreco
  float media_kms = 3;   // O Go vai ler isto como MediaKms
}

message LocalizacaoStats {
  int32 total_carros = 1;
  float valor_total = 2;
}